synapse/storage/events.py

```diff
@@ -610,6 +594,42 @@ def event_dict(event):
             ],
         )
 
+        if event.type == EventTypes.Redaction and event.redacts is not None:
+            self._remove_push_actions_for_event_id_txn(
+                txn, event.room_id, event.redacts
+            )
+
+        self._store_mult_state_groups_txn(txn, events_and_contexts)
+
+        self._handle_mult_prev_events(
+            txn,
+            events=[event for event, _ in events_and_contexts],
+        )
+
+        for event, _ in events_and_contexts:
+            if event.type == EventTypes.Name:
```

Comment:
I'm not sure I can reasonably factor it out since it is such a mess in it's current state.
Note the wonderfully inconsistent mixture of tables being interfered with here.